set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -371,42 +349,33 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
 	 * inserts item within user set to given position or at the end if no valid position provided.
 	 * 
 	 * @param request
-	 * @param wskey The API key
 	 * @param identifier The identifier of a user set
 	 * @param datasetId The identifier of the dataset, typically a number
 	 * @param localId The local identifier within the provider
 	 * @param position The position in the existin item list
-	 * @param userSet The user set fields to update in JSON format e.g. title or description
 	 * @param profile The profile definition
 	 * @param action The action describing the request
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	protected ResponseEntity<String> insertItemIntoUserSet(HttpServletRequest request, String wsKey, 
-			String identifier, String datasetId, String localId, String position, String userToken, 
+	protected ResponseEntity<String> insertItemIntoUserSet(HttpServletRequest request,  
+			String identifier, String datasetId, String localId, String position,  
 			String profileStr, String action) throws HttpException {
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			verifyWriteAccess(Operations.UPDATE, request);
```

Comment:
this method needs to be invoked in the calling method

Reply:
verifyWriteAccess invoked as suggested